llvm/lib/Target/AArch64/AArch64InstrInfo.td

```diff
@@ -6663,6 +6663,42 @@ defm : SIMDAcrossLanesUnsignedIntrinsic<"UMINV", AArch64uminv>;
 def : Pat<(v2i32 (AArch64uminv (v2i32 V64:$Rn))),
           (UMINPv2i32 V64:$Rn, V64:$Rn)>;
 
+// For vecreduce_{opc}
```

Comment:
Can you add a comment explaining that these patterns are used by GlobalISel. It's a bit of a shame that we can't get SDAG to use the same nodes, but the return types can make that awkward. It works nicer in GlobalISel where we can opt for the return type to be in a FPR.